pkg/skaffold/errors/err_map.go

```diff
@@ -31,28 +31,59 @@ func re(s string) *regexp.Regexp {
 
 type problem struct {
 	regexp      *regexp.Regexp
-	description string
+	description func(error) string
```

Comment:
Perhaps we're doing the same thing in two different ways: we match `regexp` against error to filter down to the `problem` but now we match error again to filter down to a description.

We can leave this as is and rather split `DockerConnectionFailedPrefix` into `DockerConnectionFailed` and `DockerConnectionFailedWithUnknownHost` for the case with known host address.
That'll keep `problem` struct at lowest granularity and easily referenceable. 

WDYT?

Reply:
The above approach still won't solve the problem. The reason why i added the `description` as a function was to extract the "address" where docker daemon should be running from the error message. 
e.g. if the error is 
```
exiting dev mode because first build failed: docker build: Cannot connect to the Docker daemon at tcp://127.0.0.1:32770. Is the docker daemon running?
```
I want to extract the "tcp://127.0.0.1:32770." and trim the error message as
```
Build failed. Cannot connect to the Docker daemon at tcp://127.0.0.1:32770. Is the docker daemon running?
```
When deploy context is not minikube, we could just return the error we receive as is.

When deploy context is minikube, the error we receive is
```
creating runner: creating builder: getting docker client: getting minikube env: running [/Users/tejaldesai/Downloads/google-cloud-sdk2/bin/minikube docker-env --shell none -p minikube]
 - stdout: "\n\n"
 - stderr: "! Executing \"docker container inspect minikube --format={{.State.Status}}\" took an unusually long time: 7.36540945s\n* Restarting the docker service may improve performance.\nX Exiting due to GUEST_STATUS: state: unknown state \"minikube\": docker container inspect minikube --format=: exit status 1\nstdout:\n\n\nstderr:\nCannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\n\n* \n* If the above advice does not help, please let us know: \n  - https://github.com/kubernetes/minikube/issues/new/choose\n"
 - cause: exit status 80
```

which needs to be translated to 
```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

Another way to do this, if we use parse the "stdout" when we get this error from minikube and raise the error as "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"
maybe this is long term approach we want. 

Or, else we use a static string for both cases.
```
Cannot connect to the Docker daemon. Is the docker daemon running?
```

I hope it's clear.

In general, i feel we need a function to massage/trim the error we received to get rid of all the text added due to `errors.Wrap`. 

Or, we could add another field to  struct problem 
```
type problem struct {
	regexp      *regexp.Regexp
	description func(error) string
	errCode     proto.StatusCode
	suggestion  func(opts config.SkaffoldOptions) []*proto.Suggestion
        userMessage  func(error) string
}
```